File: src/main/java/templemore/onx/version1/OandX.java
package templemore.onx.version1;

/**
 * @author Chris Turner
 */
public class OandX {

    public static void main(String[] args) {
        final Grid grid = new Grid();
        final WinScanner winScanner = new WinScanner();
        final MoveFinder moveFinder = new MoveFinder();
        Token token = Token.nought;

        while ( !grid.isFull() && !winScanner.isWin(grid) ) {
            System.out.println("Player: " + token.getSymbol());
            grid.addToken(moveFinder.findBestPosition(grid, token), token);
            System.out.println(grid.toString());
            token = token.flip();
        }

        if ( winScanner.isWin(grid) ) {
            System.out.println("Game won by: " + winScanner.winningToken(grid).getSymbol());
        }
        else {
            System.out.println("Game drawn");
        }
    }
}

File: src/main/scala/templemore/onx/version6/OandX.scala
package templemore.onx.version6

/**
 * @author Chris Turner
 */
object OandX {

  def main(args: Array[String]) {
    Grid.randomness = true

    val grid = new GridActor().start
    val noughts = new PlayerActor(Nought).start
    val crosses = new PlayerActor(Cross).start
    val game = new GameActor(grid, noughts, crosses).start

    game ! StartGame
  }
}